apps/aechannel/src/aesc_fsm.erl

```diff
@@ -3376,17 +3378,12 @@ handle_common_event_(cast, {?CHANNEL_CLOSING, Info} = Msg, _St, _, D) ->
             lager:debug("Channel solo-closing", []),
             report_on_chain_tx(solo_closing, SignedTx, D1),
             next_state(channel_closing, D1);
-        {ok, solo_closed, SignedTx} ->
-            lager:debug("Channel solo-closed", []),
-            report_on_chain_tx(solo_closed, SignedTx, D1),
-            close(channel_no_longer_active, D1);
-        {error, channel_active} ->
-            %% Weird, but possible e.g. due to forking (TODO: what might happen next?)
-            keep_state(log(drop, msg_type(Msg), Msg, D));
-        {error, _} = Error ->
-            %% Couldn't find the channel object on-chain, or something weird happened
-            lager:debug("Channel not found, or other: ~p", [Error]),
-            close(channel_no_longer_active, D)
+        {error, not_solo_closing} ->
+            %% the min depth watcher reported a channel object that is not
+            %% closing, this could be due to a (micro) fork that rejected the
+            %% channel_close_solo_tx that had been initally detected
+            lager:debug("Received a channel closing event for not closing channel", []),
+            keep_state(D1)
```

Comment:
Previously this was doing `close(channel_no_longer_active, D)` that was stopping the FSM.

Does the `latest` (or other) field of the state of the FSM (or of other process?) need update for keeping some invariants? (I admit I do not have full understanding of the related state: based on quick look, I did not find evident reasons for the `latest` field to be updated.)

Reply:
This is a curious case when the min depth watcher reports a channel is closing while it is really not (due to a _rapid_ change on-chain, possibly a fork). This is basically catching a misleading incomming event, hence no change to `latest`